workbench-core/example/infrastructure/integration-tests/support/resources/datasets/datasets.ts

```diff
@@ -6,31 +6,75 @@ import { AxiosResponse } from 'axios';
 import ClientSession from '../../clientSession';
 import RandomTextGenerator from '../../utils/randomTextGenerator';
 import CollectionResource from '../base/collectionResource';
-import Dataset from './dataset';
+import Dataset, { DataSetCreateParams } from './dataset';
 
 export default class Datasets extends CollectionResource {
   public constructor(clientSession: ClientSession) {
     super(clientSession, 'datasets', 'dataset');
     this._api = 'datasets';
   }
 
-  public dataset(id: string): Dataset {
-    return new Dataset(id, this._clientSession, this._api);
+  public dataset(params: DataSetCreateParams): Dataset {
+    return new Dataset(params);
   }
 
-  public async import(requestBody: { [id: string]: string }): Promise<AxiosResponse> {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  public async create(body: any = {}, applyDefault: boolean = true): Promise<AxiosResponse> {
+    // Because of the cleanup logic, before we do the create, we need to ensure that the extender of this collection
+    // resource class has a method that returns the resource operations helper for the child resource.
+    // For example, if the extender class is 'Users' and it provides childType = 'user', then Users class must have
+    // a method called 'user()'.
+    const requestBody = applyDefault ? this._buildDefaults(body) : body;
+    const response: AxiosResponse = await this._axiosInstance.post(this._api, requestBody);
+
+    const createParams: DataSetCreateParams = {
+      id: response.data.id,
+      clientSession: this._clientSession,
+      parentApi: 'datasets',
+      awsAccountId: response.data.awsAccountId,
+      storageName: response.data.storageName,
+      storagePath: response.data.path
+    };
+    const taskId = `${this._childType}-${createParams.id}`;
+    const resourceNode = this.dataset(createParams);
+    this.children.set(resourceNode.id, resourceNode);
+    // We add a cleanup task to the cleanup queue for the session
+    this._clientSession.addCleanupTask({ id: taskId, task: async () => resourceNode.cleanup() });
+
+    return response;
+  }
+
+  // List call
+  public async get(queryParams?: Record<string, string>): Promise<AxiosResponse> {
+    if (!queryParams) {
+      return this._axiosInstance.get(this._api, { params: queryParams });
+    }
+    return this._axiosInstance.get(`${this._api}/${queryParams.id}`);
+  }
+
+  public async delete(queryParams: Record<string, string>): Promise<AxiosResponse> {
+    return this._axiosInstance.delete(`${this._api}/${queryParams.id}`);
+  }
```

Comment:
I think these belong in `dataset.ts`, not here since they are operating on individual datasets, not the collection of datasets

Reply:
Nope. Can't move these. It seems sensible and we can totally use the existing get/delete apis on base/resource that way, but these are here specifically so we can work with fake ids for error path testing.